src/datastructures/Classes.java

```diff
@@ -44,13 +42,27 @@ public boolean createClass(String name) {
 
 	public boolean renameClass(String target, String newname) {
 		int index = entities.indexOf(new Entity(target));
-		if (index >= 0) {
-			entities.get(index).setName(newname);
-			return true;
-		} else {
+
+		if (index < 0) {
 			// Target not found.
 			return false;
 		}
+		// Changing entity name.
+		entities.get(index).setName(newname);
+
+		// Changing entity's name in relationships.
+		for (Relationship r : relationships) {
+			if (r.getName().equals(target)) {
+				r.setName(newname);
+			}
```

Comment:
Do we need this? It seems that it is looking for a relationship with the same name as the class to be renamed, and renaming the relationship. Unless we want to rename relationships that are named after an outdated class to avoid confusion, I feel that the two if statements below would suffice.

Reply:
You're right! This shouldn't need to be checked, my bad!